src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -195,6 +197,32 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             });
         }
 
+        private static ImmutableArray<DiagnosticInfo> GetDiagnosticsForInvalidMembers(INamedTypeSymbol symbol)
+        {
+            ImmutableArray<DiagnosticInfo>.Builder diagnostics = ImmutableArray.CreateBuilder<DiagnosticInfo>();
+            foreach (ISymbol member in symbol.GetMembers())
+            {
+                if (member.IsStatic)
+                {
+                    continue;
+                }
+
+                DiagnosticDescriptor? descriptor = member.Kind switch
+                {
+                    SymbolKind.Property => GeneratorDiagnostics.InstancePropertyDeclaredInInterface,
+                    SymbolKind.Event => GeneratorDiagnostics.InstanceEventDeclaredInInterface,
+                    _ => null
+                };
+
+                if (descriptor is not null)
+                {
+                    diagnostics.Add(member.CreateDiagnosticInfo(descriptor, member.Name, symbol.ToDisplayString()));
+                }
```

Comment:
Could we extract this into a method in ComMethodInfo.cs and call it where we loop over the members in `GetMethodFromInterface` (and we could maybe rename that method).